src/foundry/foundry.js/config.d.ts

```diff
@@ -1811,175 +1400,123 @@ declare global {
-         */
-        foreground: ConstructorOf<ForegroundLayer>;
-
-        /**
-         * @defaultValue `SoundsLayer`
-         */
-        sounds: ConstructorOf<SoundsLayer>;
-
-        /**
-         * @defaultValue `LightingLayer`
-         */
-        lighting: ConstructorOf<LightingLayer>;
-
-        /**
-         * @defaultValue `SightLayer`
-         */
-        sight: ConstructorOf<SightLayer>;
-
-        /**
-         * @defaultValue `EffectsLayer`
-         */
-        effects: ConstructorOf<EffectsLayer>;
-
-        /**
-         * @defaultValue `ControlsLayer`
-         */
-        controls: ConstructorOf<ControlsLayer>;
-
-        [key: string]: ConstructorOf<CanvasLayer>;
+        /** @defaultValue `{ layerClass: BackgroundLayer, group: "primary" }` */
+        background: LayerDefinition<ConstructorOf<BackgroundLayer>>;
+
+        /** @defaultValue `{ layerClass: DrawingsLayer, group: "primary" }` */
+        drawings: LayerDefinition<ConstructorOf<DrawingsLayer>>;
+
+        /** @defaultValue `{ layerClass: GridLayer, group: "primary" }` */
+        grid: LayerDefinition<ConstructorOf<GridLayer>>;
+
+        /** @defaultValue `{ layerClass: TemplateLayer, group: "primary" }` */
+        templates: LayerDefinition<ConstructorOf<TemplateLayer>>;
+
+        /** @defaultValue `{ layerClass: TokenLayer, group: "primary" }` */
+        tokens: LayerDefinition<ConstructorOf<TokenLayer>>;
+
+        /** @defaultValue `{ layerClass: ForegroundLayer, group: "primary" }` */
+        foreground: LayerDefinition<ConstructorOf<ForegroundLayer>>;
+
+        /** @defaultValue `{ layerClass: WallsLayer, group: "effects" }` */
+        walls: LayerDefinition<ConstructorOf<WallsLayer>>;
+
+        /** @defaultValue `{ layerClass: LightingLayer, group: "effects" }` */
+        lighting: LayerDefinition<ConstructorOf<LightingLayer>>;
+
+        /** @defaultValue `{ layerClass: WeatherLayer, group: "effects" }` */
+        weather: LayerDefinition<ConstructorOf<CanvasLayer>>; // FIXME: WeatherLayer
+
+        /** @defaultValue `{ layerClass: SightLayer, group: "effects" }` */
+        sight: LayerDefinition<ConstructorOf<SightLayer>>;
+
+        /** @defaultValue `{ layerClass: SoundsLayer, group: "interface" }` */
+        sounds: LayerDefinition<ConstructorOf<SoundsLayer>>;
+
+        /** @defaultValue `{ layerClass: NotesLayer, group: "interface" }` */
+        notes: LayerDefinition<ConstructorOf<NotesLayer>>;
+
+        /** @defaultValue `{ layerClass: ControlsLayer, group: "interface" }` */
+        controls: LayerDefinition<ConstructorOf<ControlsLayer>>;
+
+        [key: string]: LayerDefinition<ConstructorOf<CanvasLayer>>;
+      }
+
+      // TODO: The type spec could be improved, if we add a types only interface common to the canvas groups. Otherwise
+      // the first common superclass they share is `PIXI.Container`.
+      interface GroupDefinition<GroupClass extends ConstructorOf<PIXI.Container>> {
```

Comment:
Hm, it might make sense to have interfaces both for the static side and the instance side. Something like
```ts
interface CanvasGroup extends PIXI.Container {
  sortableChildren: boolean;

  // potentially, _createLayers could be added, but it should be protected / @internal but that's not possible in an interface... So I think we should leave it out, it's only there by convention.
}

type PixiContainerConstructor = typeof PIXI.Container;

interface CanvasGroupConstructor extends PixiContainerConstructor {
  new (): CanvasGroup;

  /**
   * The name of this canvas group
   */
  groupName: string;
}
```

Reply:
I will add those in the canvas groups PR.